subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -353,13 +353,7 @@ public TransformServiceLookup(Provider<FileSystemLocation> inputFileProvider, @N
             this.delegate = delegate;
             ImmutableList.Builder<InjectionPoint> builder = ImmutableList.builder();
             builder.add(InjectionPoint.injectedByAnnotation(InputArtifact.class, File.class, () -> {
-                DeprecationLogger
-                    .deprecate("Injecting the input artifact of a transform as a File")
-                    .withAdvice("Declare the input artifact as Provider<FileSystemLocation> instead.")
-                    .willBeRemovedInGradle7()
-                    .withUserManual("artifact_transforms", "sec:implementing-artifact-transforms")
-                    .nagUser();
-                return inputFileProvider.get().getAsFile();
+                throw new InvalidUserCodeException("Injecting the input artifact of a transform as a File isn't allowed. Declare the input artifact as Provider<FileSystemLocation> instead.");
```

Comment:
Instead of changing the exception here, we should remove `supportedTypes = { File.class }` from the `@InjectionPointQualifier` on `@InputArtifact`.

Reply:
mmm, good point I wasn't aware of this :+1: